src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs

```diff
@@ -542,9 +542,10 @@ private void SetProxyCredentials(NetworkCredential credentials)
 
                     // Unlike normal credentials, proxy credentials are URL decoded by libcurl, so we URL encode 
                     // them in order to allow, for example, a colon in the username.
+                    string userNameColonPassword = WebUtility.UrlEncode(credentials.UserName) + ":" + WebUtility.UrlEncode(credentials.Password);
                     string credentialText = string.IsNullOrEmpty(credentials.Domain) ?
-                        string.Format("{0}:{1}", WebUtility.UrlEncode(credentials.UserName), WebUtility.UrlEncode(credentials.Password)) :
-                        string.Format("{2}\\{0}:{1}", WebUtility.UrlEncode(credentials.UserName), WebUtility.UrlEncode(credentials.Password), WebUtility.UrlEncode(credentials.Domain));
+                        userNameColonPassword :
+                        WebUtility.UrlEncode(credentials.Domain) + "\\" + userNameColonPassword;
```

Comment:
Welcome back, Steve! 😃

> An improvement in what measurement?

I was just measuring time and found that the two `String.Concat` calls is faster than `String.Format`, despite the extra string allocation.

However, after your question and thinking about it more, I decided to switch the else case back to `String.Format` (commit updated). It's slightly slower, but I prefer avoiding the extra string allocation.

Here's a simple benchmark:

```c#
using System;
using System.Diagnostics;

public class Program
{
    public static void Main()
    {
        const string username = "username";
        const string password = "password";
        const string domain = "domain";

        TimeAction("  Old", () => Old(username, password, domain));
        TimeAction("New 1", () => New1(username, password, domain));
        TimeAction("New 2", () => New2(username, password, domain));
    }

    private static string Old(string username, string password, string domain)
    {
        return string.Format("{2}\\{0}:{1}", username, password, domain);
    }

    private static string New1(string username, string password, string domain)
    {
        string usernamePassword = username + ":" + password;
        return domain + "\\" + usernamePassword;
    }

    private static string New2(string username, string password, string domain)
    {
        return domain + "\\" + username + ":" + password;
    }

    private static void TimeAction(string prefix, Action action, int times = 5, int iterations = 10000000)
    {
        var sw = new Stopwatch();
        for (int i = 0; i < times; i++)
        {
            int gen0 = GC.CollectionCount(0);
            sw.Restart();
            for (int iter = 0; iter < iterations; iter++)
            {
                action();
            }
            sw.Stop();
            Console.WriteLine($"{prefix}: Time: {sw.Elapsed.TotalSeconds}\tGC0: {GC.CollectionCount(0) - gen0}");
        }
        Console.WriteLine("-----------------");
    }
}
```

Results:

```
  Old: Time: 1.3867583  GC0: 190
  Old: Time: 1.232599   GC0: 191
  Old: Time: 1.2480056  GC0: 191
  Old: Time: 1.2423305  GC0: 190
  Old: Time: 1.330423   GC0: 191
-----------------
New 1: Time: 0.4455669  GC0: 344
New 1: Time: 0.5790028  GC0: 343
New 1: Time: 0.4898952  GC0: 343
New 1: Time: 0.4232398  GC0: 343
New 1: Time: 0.4240736  GC0: 344
-----------------
New 2: Time: 0.5126726  GC0: 343
New 2: Time: 0.5074243  GC0: 343
New 2: Time: 0.4884827  GC0: 344
New 2: Time: 0.5038214  GC0: 343
New 2: Time: 0.5019208  GC0: 343
```